pkg/controller/machine/controller.go

```diff
@@ -422,21 +422,26 @@ func isInvalidMachineConfigurationError(err error) bool {
 func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, errorMessage string) error {
 	if stringPointerDeref(machine.Status.Phase) != phase {
 		klog.V(3).Infof("%v: going into phase %q", machine.GetName(), phase)
+		// We are required to Patch annotations before updating the status because Machines have a status subresource.
```

Comment:
"A call to Patch will mutate our local copy of the machine to match what is stored in the API.  Before we make any changes to the status subresource on our local copy, we need to patch the object first, otherwise our local changes to the status subresource will be lost."